src/gems/simulation/extra_output.py

```diff
@@ -0,0 +1,152 @@
+# extra_output.py
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, Optional
+
+from gems.expression import evaluate
+from gems.expression.evaluate import EvaluationError, ValueProvider
+from gems.simulation.optimization import OptimizationProblem
+from gems.simulation.output_values_base import BaseOutputValue
+from gems.study.data import ComponentParameterIndex, TimeScenarioIndex
+
+
+@dataclass
+class ExtraOutput(BaseOutputValue):  # <-- INHERITS from the Base Class
+    """
+    Stores evaluated outputs (from ExpressionNodes), not solver variables.
+    Inherits all common fields (_name, _value, _size, ignore) and methods
+    (__eq__, is_close, __str__, value property) from BaseOutputValue.
+    """
+
+    def _set(
+        self,
+        timestep: Optional[int],
+        scenario: Optional[int],
+        value: float,
+        status: Optional[str] = None,  # conservé pour compatibilité avec l'interface
```

Comment:
Ah I did not pay attention that arguments were not the same... Then it may be better not to define it in the parent class to have a natural signature in `ExtraOutput`...